x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -90,6 +91,108 @@ export async function getAgents(esClient: ElasticsearchClient, options: GetAgent
   return agents;
 }
 
+export async function getAgentsByKueryPit(
+  esClient: ElasticsearchClient,
+  options: ListWithKuery & {
+    showInactive: boolean;
+    pitId: string;
+    searchAfter?: SortResults;
+  }
+): Promise<{
+  agents: Agent[];
+  total: number;
+  page: number;
+  perPage: number;
+}> {
+  const {
+    page = 1,
+    perPage = 20,
+    sortField = 'enrolled_at',
+    sortOrder = 'desc',
+    kuery,
+    showInactive = false,
+    showUpgradeable,
+    searchAfter,
+  } = options;
+  const { pitId } = options;
+  const filters = [];
+
+  if (kuery && kuery !== '') {
+    filters.push(kuery);
+  }
+
+  if (showInactive === false) {
+    filters.push(ACTIVE_AGENT_CONDITION);
+  }
+
+  const kueryNode = _joinFilters(filters);
+  const body = kueryNode ? { query: toElasticsearchQuery(kueryNode) } : {};
```

Comment:
I'm not sure if performance is too much of a concern here, ignore this if not. Do you have an example of the query this generates if kuery and showInactive are present? 

I am wondering if it puts them in a `bool.filter` block or `bool.must`, I only ask because we don't really care about scoring the results here, so there could be a performance gain if we built a filter query ourselves.

Reply:
I checked and the it uses a `bool.filter`:
```
{
    "bool": {
        "filter": [
            {
                "bool": {
                    "should": [
                        {
                            "match_phrase": {
                                "policy_id": "a1f56390-f13f-11ec-a141-5d852147df32"
                            }
                        }
                    ],
                    "minimum_should_match": 1
                }
            },
            {
                "bool": {
                    "should": [
                        {
                            "match": {
                                "active": true
                            }
                        }
                    ],
                    "minimum_should_match": 1
                }
            }
        ]
    }
}
```